### DISCUSSION TITLE
Heating with backup heater -

### DISCUSSION THREAD

**Discussion by d3zit** [raomin/ESPAltherma#288] [Q&A]
Do you know which parameters allow you to understand if the heat pump is heating the water with the help/or only with the backup heater?

If operation_mode is **Heating** and 3_way_valve is **ON** then the heatpump is heating up DHW 
If operation_mode is **Heating**,  2_way_valve is **ON**  and 3_way_valve is **OFF** is heating the water for heating system

Apparently when the system is heating the water with the backup heater it's not in one of the previous states.

Thanks!

**Comment by Dennis3321** [comment]
good question, I'd like to know that too...

**Comment by Freeman656** [comment]
Hi !
Very good question :)
@raomin any idea ?

**Comment by raomin** [comment]
I'm using these:

BUH Step1
BUH Step2

**Reply by Freeman656** [reply]
Do you know why there are 2 values ? In case of two heat resistances in some models ?

**Reply by raomin** [reply]
2 power levels. Something like 1kw and 2 kw. Might 2 resistances...

**Comment by d3zit** [comment]
![image](https://github.com/raomin/ESPAltherma/assets/72941345/601961f6-c662-45c4-ab65-6f77c29731e6)
In this graph I plotted:

Yellow: power consumption heat pump (shelly clamp)
Blue: power consumption of the heat pump resistance (another shelly clamp).
Orange: OAI (outside air temperature).
Green: DHW temperature

Resistance starts randomly and I'm unable to detect it as - apparently - when it starts the heatpump is not in heating mode.
Gray: standby 
Blue: DHW heating
Green: cooling

Heating with resistance consumption ends up in the standby because I'm unable to correctly detect the status and address the energy consumption accordingly
![image](https://github.com/raomin/ESPAltherma/assets/72941345/abfac8d7-22cd-4204-9c39-15c83974de78)

**Reply by raomin** [reply]
Add these to your graph:
BUH Step1
BUH Step2

It should show the resistance activity

**Comment by Freeman656** [comment]
I added the BUH step 1 and 2 in my HA config and I get the following data:
Strange to see that the BUH step 1 gets activated during a few minutes as it's almost 30° outside...

![image](https://github.com/raomin/ESPAltherma/assets/60422888/6c18a840-1036-4bd7-bbcf-0a77b180fb18)

**Reply by raomin** [reply]
The HP, low temp, cannot probably heat more than 50 even with 30 outside.
@KalaNagHTD probably knows this...

**Comment by d3zit** [comment]
In my case BSH is activated instead of BUH
![image](https://github.com/raomin/ESPAltherma/assets/72941345/ff99af82-b150-44e5-9603-9ec35135c661)

![image](https://github.com/raomin/ESPAltherma/assets/72941345/c40719ac-b96c-4b60-95e9-9df7eb4d5705)

I can confirm that the resistance is using approx. 1 kW
The heatpump waited approx. 30mins before starting the BSH

My OAT sensor is placed right behind the fan of the external unit. As soon as the fan starts spinning the temperature drops and therefore (I suppose) the Heatpump is tricked into turning on the resistance (??)

Unfortunately it seems that after some time even if IU operation is DHW and BSH are on, resistance is actually off as it doesn't drain power and therefore I'm "unable" to identify when the heatpump is heating with the resistance.

Later on IU operation is DHW, BSH is off.. but the resistance is back on. I know it only because I have a clamp on the resistance circuit, otherwise from Altherma I would be unable to detect this status.

**Reply by mx-vatte** [reply]
BSH is booster heater inside the DHW boiler.
BUH is backup heater inside the outdoor unit of the heat pump (correct me if I'm wrong) also used for defrost and outside temperatures below the operating area of the heat pump.

**Reply by Freeman656** [reply]
So why would the BUH start when it's very hot outside ? Could this be a misconfiguration somewhere ?
I'll check the status of the BSH next time to add some data

**Reply by hsteweri** [reply]
This is normal behaviour.
Heatpump is disabled when it's too hot outside (I think at 35°C) to prevent damage to the system.
DHW is generated by BUH in that case.
"Workaround" would be to somehow cooling the heat-sensor ;)

**Reply by Freeman656** [reply]
OK, I'll check today, it's a bit cloudy, around 20°, so it should not be too hot.

**Reply by mx-vatte** [reply]
Maybe you don't have a BSH and your config enables the BUH to support for DHW heating/keeping warm?

**Reply by mx-vatte** [reply]
But don't ignore the risks while trick the heat pump through a cooled sensor.

**Reply by Freeman656** [reply]
I won't do it anyways !! I just want to ensure everything is working as expected :)

**Reply by d3zit** [reply]
Daikin confirmed today that by default heatpump should stop  working at 25, but should be configurable at 35°C

**Reply by d3zit** [reply]
I also asked if it's possible to move the sensor somewhere else. My heatpump is mounted facing west and therefore after 1pm it sits in plain sunlight reading values much higher then real ones. I'll try to design and build a 3D cage with a small fan to provide continuous active ventilation to the sensor in order to read the real air temperature

**Reply by Freeman656** [reply]
Really interesting !

**Reply by Freeman656** [reply]
Do you know which parameter it is ?

**Comment by Freeman656** [comment]
BSH didn't go up, only BUH for a few minutes.
![image](https://github.com/raomin/ESPAltherma/assets/60422888/b1ef4f3f-0f0f-49a9-8760-8de026a33706)

**Reply by mx-vatte** [reply]
Are you're sure that you're using the right register configuration regarding your heat pump? Maybe BSH and BSU are switched.

**Reply by d3zit** [reply]
No. Next month a daikin technician should have a look and let you know.  I don't even know how a BUH should look like... do you have photos?

Anyway I can confirm that 25°C is the threshold

**Reply by mx-vatte** [reply]
This should be the setting [4-02]. After my understanding and observations this is a limit for heating not DHW. I switched it to 19 °C since I don't want that the pump is running every ~3 minutes for ~3 minutes. So over 19 °C outside temp the pump stays of as long the thermostat is off also. But the DHW is heated by the heat pump without any electrical heater (BUH nor BSH).

**Comment by Freeman656** [comment]
In ESPAlterma config I have : 
```
{0x60,12,305,1,-1,"BSH"},
{0x60,12,304,1,-1,"Palier1 BUH"},
{0x60,12,303,1,-1,"Palier2 BUH"},
```

And in my heatpump configuration, I don't really know where to look at :)

**Reply by bzq-pl** [reply]
This is good example. Can anybody explain how interpret definition of registers?

When I read above 3 registers then I see all three are under the same address (0x60) and they have the same offset (12). Difference is only in conversion but data for conversion is the same in all those 3 cases. Is it possible to store 3 different information under one address and offset?

**Reply by mx-vatte** [reply]
You can check it in the installer menu of your heat pump. Attention this is out of the documentation for my heat pump (Daikin Altherma ERLQ011CAW1 | EHBH11CB3V). There can/will be differences if you have a different heat pump. Also some settings depend on each other. 

[E-03] Steps/level of the BSH/BUH
           0: no BUH
           1: 1 level
           2: 2 level
[5-0D] BSH/BUH type (connection)
           0: 1P,(1/2)
           1: 1P,(1/1+2)
           2: 3P,(1/2)
           3: 3P,(1/1+2) (*3)
           4: 3PN,(1/2)
           5: 3PN,(1/1+2) (*4)
[6-02] BSH/BUH power in kW
[6-03] BSH/BUH power level 1 in kW
[6-04] BSH/BUH power level 2 in kW
[4-00] BSH/BUH operating mode
          0: disabled
          1: enabled
          2: only DHW
[4-01]: BSH/BUH priority
          0: no
          1: BSH
          2: BUH
[4-03] BSH/BUH operating permit
          0: limited
          1: no limit
          2: always optimal
          3: optimal
[4-07] BSH/BUH level 2 activated?
           0: no
           1: yes
[5-01] BSH/BUH enabling outside temperature in °C

**Reply by bzq-pl** [reply]
What do you mean by "installer menu"? Are you referring to menu in heat pump after enabling installer profile or are you referring to some documentation? I have been many times in "installer mode" but I haven't seen such information. I have ground heat pump but it shouldn't be a difference.

**Reply by mx-vatte** [reply]
Yes I mean the to the installer menu in the indoor operation unit after enabling the installer mode.
Should look like this...
![P_20221007_092617](https://github.com/raomin/ESPAltherma/assets/123895372/5c842e62-6feb-44a9-b83b-60301e47fab2)

**Reply by bzq-pl** [reply]
thank you! need to explore my menu!

**Reply by mx-vatte** [reply]
Do you have the corresponding "Installer Reference Guide" for you heat pump? On the last pages there is an overview table of all settings.
I went through and noted all settings from my heat pump settings overview in installer mode.

**Reply by bzq-pl** [reply]
https://www.daikin-ce.com/content/dam/document-library/Installer-reference-guide/heat/ground-to-water-heat-pump/EGSAH-D9W,EGSAX-D9WG_4PEN569820-1B_Installer%20reference%20guide_English.pdf

I have looked at the chapter "Field settings table" but I'm not sure if this is the data I should explore.

**Reply by mx-vatte** [reply]
This is the chapter/table which I went through. But some settings became clear for me only in connection to the explanations inside the manual. Therefore the search function (field code) inside the PDF is really helpful.

**Reply by bzq-pl** [reply]
thank you - I need to explore this document in the way you proposed.
